### PR TITLE
python310: Update to version 3.10.6

### PR DESCRIPTION
**Pull request by th0ma7**
## Description

python310: Update to version 3.10.6

NOTE: `numpy` fails to build using setuptools >= 64.x

## Checklist

- [ ] Build rule `all-supported` completed successfully
- [ ] New installation of package completed successfully
- [ ] Package upgrade completed successfully (Manually install the package again)
- [ ] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Update-Policy#tests-checks)
- [ ] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created

### Type of change

- [ ] Bug fix
- [ ] New Package
- [x] Package update
- [ ] Includes small framework changes
- [ ] This change requires a documentation update (e.g. Wiki)

### CODE REVIEW

File: cross/python310/Makefile [hunk 1]
```diff
@@ -130,10 +130,10 @@ python310_post_install: $(WORK_DIR)/python-cc.mk
...
-	. $(WORK_DIR)/crossenv/bin/activate && $(RUN) build-python get-pip.py "pip==22.0.4" --no-setuptools --no-wheel --disable-pip-version-check
-	. $(WORK_DIR)/crossenv/bin/activate && $(RUN) build-pip --disable-pip-version-check install "setuptools==62.4.0" "setuptools-scm==6.4.2" "wheel==0.37.1" "cffi==1.15.0" "cryptography==3.3.2" "poetry==1.1.13" "Cython==0.29.28" "flit==3.7.1"
-	. $(WORK_DIR)/crossenv/bin/activate && $(RUN) python get-pip.py "pip==22.0.4" --no-setuptools --no-wheel --disable-pip-version-check
-	. $(WORK_DIR)/crossenv/bin/activate && $(RUN) pip --disable-pip-version-check install "setuptools==62.4.0" "setuptools-scm==6.4.2" "wheel==0.37.1" "cffi==1.15.0" "cryptography==3.3.2" "poetry==1.1.13" "Cython==0.29.28" "flit==3.7.1"
+	. $(WORK_DIR)/crossenv/bin/activate && $(RUN) build-python get-pip.py "pip==22.2.2" --no-setuptools --no-wheel --disable-pip-version-check
+	. $(WORK_DIR)/crossenv/bin/activate && $(RUN) build-pip --disable-pip-version-check install "setuptools==63.4.3" "setuptools-scm==7.0.5" "wheel==0.37.1" "cffi==1.15.1" "cryptography==3.3.2" "poetry==1.1.14" "Cython==0.29.32" "flit==3.7.1"
```
Comment (hgy59) [inline cross/python310/Makefile:134]: shall we install `setuptools-rust` here, to prepare the build of cryptography wheel with rust?

**Comment by hgy59** [comment]
@th0ma7 see comment in https://github.com/numpy/numpy/issues/22135#issuecomment-1215456840, numpy has pinned setuptools to v59.2.0 in their environment. So this is probably not a but.

**Comment by th0ma7** [comment]
> @th0ma7 see comment in [numpy/numpy#22135 (comment)](https://github.com/numpy/numpy/issues/22135#issuecomment-1215456840), numpy has pinned setuptools to v59.2.0 in their environment. So this is probably not a bug, but not supported by intention.

Good catch.  They actually mention <=64.x https://github.com/numpy/numpy/issues/22135#issuecomment-1216567894

File: cross/python310/Makefile (hunk 1 above)
Comment (th0ma7) [reply cross/python310/Makefile:134]: We probably could from a preparation perspective.  But what's missing afterwards, at least what I'm "guessing", is having a `native/rust` that translate the install portion of `spksrc.cross-rust.mk` and ensure this get taken into account when generating the `tc_vars.*" so a rustc compiler is part of the default toolchain.  Now that I'm thinking about it, the easiest may be to alway have rustc part of the toolchain portion so it is ready & available by default always.

Only then will it become feasibble to start building rust wheels.

Comment (th0ma7) [reply cross/python310/Makefile:134]: @hgy59 further thinking about it, I'd recommend leaving it aside for now until we integrate that into a rust solution.

Comment (hgy59) [reply cross/python310/Makefile:134]: I am working on it (#5399)

Comment (th0ma7) [reply cross/python310/Makefile:134]: great!  then I'll leave it out from this PR.

**Comment by th0ma7** [comment]
This was tested and LGTM.  @hgy59 seing anything missing before I merge & shoule I wait for #5399 before publishing?

**Comment by hgy59** [comment]
> This was tested and LGTM. @hgy59 seing anything missing before I merge & shoule I wait for #5399 before publishing?

Or you could cherry pick the setuptools-rust (3262891041f12677a7ae55f823b3592e3c8392b3) from it, so #5399 will not trigger the build of all python310 dependent packages.
But I am not sure, whether it is the right way to build cryptography within crossenv. Regular builds use PyO3 and I am very confused about how pip builds cryptography with setuptools-rust and what is going on under the hood... (python is still a strange ecosystem for me...).

**Comment by th0ma7** [comment]
> cherry pick the setuptools-rust 

Will do.

> I am very confused about how pip builds cryptography with setuptools-rust and what is going on under the hood... (python is still a strange ecosystem for me...).

indeed, it's not all that easy to follow, nor am I understanding it all...

**Pull request merged**